### PR TITLE
ENT - Vue list component implementation

### PR DESCRIPTION
**Pull request by gayan-orangehrm**
This PR contains,

Developments,
1. List.vue component development.
2. Dynamic cell (Runtime cell rendering without defining them as template syntax) support as in develop branch.
3. Table Left Panel component implementation.
4. Storybook data binding.

Bug fixes,
1. When implementing the clickable behaviors, consider right click, keyboard frendly
2. Name the sidebar as leftpannel in the list component configuration json
3. All the props and events should be documented in the StoryBook
4. Styles applied for generic components inside the List.vue should be added to generic components. Profilce pic,

### CODE REVIEW

File: components/src/core/components/Button/Button.vue [hunk 1]
```diff
@@ -61,6 +71,14 @@ export default defineComponent({
     iconRightName: {
       type: String,
     },
+    tooltip: {
+      type: String,
+      default: null,
+    },
+    flow: {
+      type: String,
+      default: 'top',
```
Comment (Super-Chama) [inline components/src/core/components/Button/Button.vue:80]: This kinds of props if validated is better. check [here](https://github.com/orangehrm/oxd/blob/main/components/src/core/components/Input/Select/SelectInput.vue#L119) and [here](https://v3.vuejs.org/guide/component-props.html#prop-validation)

File: components/src/core/components/Button/button.scss [hunk 2]
```diff
@@ -148,3 +148,7 @@
     padding-right: 5px;
   }
 }
+
+.oxd-button-label-wrapper {
+  display: initial;
+}
```
Comment (Super-Chama) [inline components/src/core/components/Button/button.scss:154]: add new line

File: components/src/core/components/CardTable/CardTable.vue [hunk 3]
```diff
@@ -85,6 +85,14 @@ export default defineComponent({
       type: String,
       default: () => nanoid(8),
     },
+    styleClass: {
```
Comment (Super-Chama) [inline components/src/core/components/CardTable/CardTable.vue:88]: why not use [class attribute](https://v3.vuejs.org/guide/component-attrs.html#non-prop-attributes) directly?

File: components/src/core/components/CardTable/Cell/DefaultCellContainer.vue [hunk 4]
```diff
@@ -10,6 +10,7 @@
       :item="items[header.name]"
       :header="header"
       :rowItem="items"
+      :modelValue="items.selectedStage"
```
Comment (Super-Chama) [inline components/src/core/components/CardTable/Cell/DefaultCellContainer.vue:13]: if `modelValue` is used, you will need to emit `modeValue` change event for `v-model` to work

File: components/src/core/components/CardTable/Cell/ProfilePicCell.vue [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+import ProfilePic from '@orangehrm/oxd/core/components/ProfilePic/ProfilePic.vue';
+import {cellMixin} from './cell-mixin';
+
+export default defineComponent({
+  name: 'oxd-table-cell-profile-pic',
+  mixins: [cellMixin],
+  components: {'oxd-profile-pic': ProfilePic},
+  props: {
+    rowItem: {
+      type: Object,
```
Comment (Super-Chama) [inline components/src/core/components/CardTable/Cell/ProfilePicCell.vue:16]: if possible give type here. [check](https://github.com/orangehrm/oxd/blob/main/components/src/core/components/Topbar/Navigation.vue#L70)

File: components/src/core/components/CardTable/Cell/cell-mixin.ts [hunk 6]
```diff
@@ -3,7 +3,7 @@ import {decoratorMixin} from '../Decorator/decorator-mixin';
 
 export const cellMixin = defineComponent({
   mixins: [decoratorMixin],
-  inject: ['screenState'],
+  inject: ['tableProps', 'screenState'],
```
Comment (Super-Chama) [inline components/src/core/components/CardTable/Cell/cell-mixin.ts:6]: `tableProps` are already injected via [decorator mixin](https://github.com/orangehrm/oxd/blob/ent/components/src/core/components/CardTable/Decorator/decorator-mixin.ts#L4).

File: components/src/core/components/CardTable/types.ts [hunk 7]
```diff
@@ -32,4 +45,4 @@ export type Order = 'ASC' | 'DESC' | 'DEFAULT';
 
 export interface SortDefinition {
   [column: string]: Order;
-}
+}
```
Comment (Super-Chama) [inline components/src/core/components/CardTable/types.ts:48]: newline

File: components/src/core/components/Drawer/Drawer.vue [hunk 8]
```diff
@@ -0,0 +1,234 @@
+<template>
+  <div class="drawer-wrapper">
+    <div v-if="modalState" class="overlay"></div>
+    <div
+      class="oxd-drawer"
+      ref="wrapper"
+      :class="[
```
Comment (Super-Chama) [inline components/src/core/components/Drawer/Drawer.vue:7]: better move to computed property

File: components/src/core/components/Drawer/Drawer.vue [hunk 9]
```diff
@@ -0,0 +1,234 @@
...
+  },
+
+  props: {
+    autoAlign: {
+      type: Boolean,
+      default: true,
+    },
+    position: {
+      type: String,
+      default: 'right',
```
Comment (Super-Chama) [inline components/src/core/components/Drawer/Drawer.vue:69]: consider adding types and validations

File: components/src/core/components/Drawer/Drawer.vue [hunk 10]
```diff
@@ -0,0 +1,234 @@
...
+      default: false,
+    },
+  },
+
+  directives: {
+    'click-outside': clickOutsideDirective,
+  },
+
+  setup(props, {emit}) {
+    const open = ref(false);
```
Comment (Super-Chama) [inline components/src/core/components/Drawer/Drawer.vue:114]: all these refs can be taken to a [reactive](https://v3.vuejs.org/guide/reactivity-fundamentals.html#declaring-reactive-state) with types.

**Review by Super-Chama — CHANGES_REQUESTED** [review]
Please consider these changes. Have few more files to check, will add more comments shortly

File: components/src/core/components/Button/Button.vue (hunk 1 above)
Comment (gayan-orangehrm) [reply components/src/core/components/Button/Button.vue:80]: fixed

File: components/src/core/components/Button/button.scss (hunk 2 above)
Comment (gayan-orangehrm) [reply components/src/core/components/Button/button.scss:154]: added

File: components/src/core/components/CardTable/CardTable.vue (hunk 3 above)
Comment (gayan-orangehrm) [reply components/src/core/components/CardTable/CardTable.vue:88]: used the class attribute directly

File: components/src/core/components/CardTable/Cell/DefaultCellContainer.vue (hunk 4 above)
Comment (gayan-orangehrm) [reply components/src/core/components/CardTable/Cell/DefaultCellContainer.vue:13]: fixed

File: components/src/core/components/CardTable/Cell/ProfilePicCell.vue (hunk 5 above)
Comment (gayan-orangehrm) [reply components/src/core/components/CardTable/Cell/ProfilePicCell.vue:16]: updated

File: components/src/core/components/CardTable/Cell/cell-mixin.ts (hunk 6 above)
Comment (gayan-orangehrm) [reply components/src/core/components/CardTable/Cell/cell-mixin.ts:6]: Removed 'tableProps' and  'screenState' as both are available in decorator mixin

File: components/src/core/components/CardTable/types.ts (hunk 7 above)
Comment (gayan-orangehrm) [reply components/src/core/components/CardTable/types.ts:48]: updated

File: components/src/core/components/Drawer/Drawer.vue (hunk 8 above)
Comment (gayan-orangehrm) [reply components/src/core/components/Drawer/Drawer.vue:7]: moved to a computed property

File: components/src/core/components/Button/Button.vue (hunk 1 above)
Comment (gayan-orangehrm) [reply components/src/core/components/Button/Button.vue]: validated

File: components/src/core/components/Drawer/Drawer.vue (hunk 10 above)
Comment (gayan-orangehrm) [reply components/src/core/components/Drawer/Drawer.vue]: fixed

File: components/src/core/components/List/List.vue [hunk 11]
```diff
@@ -0,0 +1,570 @@
...
+    :class="{
+      'table-sidebar-open':
+        configurations.table.sidebar.visible && state.isSidePanelOpen,
+    }"
+  >
+    <oxd-table-sidebar
+      v-if="configurations.table.sidebar.visible"
+      class="oxd-table-sidebar"
+      :class="{'with-filters': configurations.table.topFilters.visible}"
+      width="230px"
```
Comment (Super-Chama) [inline components/src/core/components/List/List.vue:13]: `width="230px"` If this is fixed value better add in css

File: components/src/core/components/List/List.vue [hunk 12]
```diff
@@ -0,0 +1,570 @@
...
+          :pages-list="pagination.pages"
+          :current="pagination.limit"
+          @previous="previous"
+          @next="next"
+          @clickPage="clickPage"
+          @onPerPageSelect="perPageSelect"
+        />
+      </div>
+    </div>
+    <!-- <oxd-drawer
```
Comment (Super-Chama) [inline components/src/core/components/List/List.vue:130]: remove commented code

File: components/src/core/components/List/List.vue [hunk 13]
```diff
@@ -0,0 +1,570 @@
...
+
+<script lang="ts">
+import CardTable from '@orangehrm/oxd/core/components/CardTable/CardTable.vue';
+import TableFilter from '@orangehrm/oxd/core/components/TableFilter/TableFilter.vue';
+import IconButton from '@orangehrm/oxd/core/components/Button/Icon.vue';
+import Button from '@orangehrm/oxd/core/components/Button/Button.vue';
+import QuickSearchInput from '@orangehrm/oxd/core/components/Input/Autocomplete/QuickSearchInput.vue';
+import TableSidebar from '@orangehrm/oxd/core/components/TableSidebar/TableSidebar.vue';
+import ProfilePic from '@orangehrm/oxd/core/components/ProfilePic/ProfilePic.vue';
+// import Drawer from '@orangehrm/oxd/core/components/Drawer/Drawer.vue';
```
Comment (Super-Chama) [inline components/src/core/components/List/List.vue:177]: no comments please

File: components/src/core/components/List/List.vue [hunk 14]
```diff
@@ -0,0 +1,570 @@
...
+      default: () => [],
+    },
+    quickSearchOptions: {
+      type: Function,
+      required: true,
+    },
+    pagination: {
+      type: Object,
+      default: () => ({
+        limit: 10 as number,
```
Comment (Super-Chama) [inline components/src/core/components/List/List.vue:222]: `10 as number` ?

File: components/src/core/components/List/List.vue [hunk 15]
```diff
@@ -0,0 +1,570 @@
...
+      const sortableFieldsObj = {};
+      props.configurations.table.headers.forEach(header => {
+        if (header.initialSortOrder) {
+          sortableFieldsObj[header.sortField] = header.initialSortOrder;
+        }
+      });
+      return sortableFieldsObj;
+    });
+
+    const isFloat = n => {
```
Comment (Super-Chama) [inline components/src/core/components/List/List.vue:265]: some dev comment is appreciated

File: components/src/core/components/List/List.vue [hunk 16]
```diff
@@ -0,0 +1,570 @@
...
+      clickPage,
+      perPageSelect,
+      paginationLength,
+      exportBtn,
+    };
+  },
+});
+</script>
+
+<style lang="scss" scoped>
```
Comment (Super-Chama) [inline components/src/core/components/List/List.vue:400]: better move to separate style sheet

File: storybook/stories/core/components/TableSidebar/TableSidebar.story.vue [hunk 17]
```diff
@@ -0,0 +1,185 @@
...
+        style: {
+          height: '36px'
+        }
+      }"
+      @list:onSelect="selectStage"
+    >
+      <template v-slot:body>
+        <oxd-select-input-btn
+          :button="{
+            label: selectedVacancy && selectedVacancy.label ? selectedVacancy.label : selectedStage && selectedStage.label ? selectedStage.label : '',
```
Comment (Super-Chama) [inline storybook/stories/core/components/TableSidebar/TableSidebar.story.vue:18]: :open_mouth:

**Review by Super-Chama — CHANGES_REQUESTED** [review]
_components/src/core/components/CardTable/Cell/DefaultCellContainer.vue_
remove empty file

**Pull request merged**